lib/jsonapi/plugs/underscore_parameters.ex

```diff
@@ -92,7 +92,7 @@ defmodule JSONAPI.UnderscoreParameters do
       conn =
         if opts[:replace_query_params] do
           query_params = fetch_query_params(conn).query_params
-          new_query_params = JString.expand_fields(query_params, &JString.underscore/1)
+          new_query_params = replace_query_params(query_params)
```

Comment:
Should this logic not apply to the `new_params` below?

Reply:
Sorry - I don't understand the question.  I created this `replace_query_params` function because the logic was becoming too complex to in-line here.

Reply:
My question is whether you intended for the behavior of this plugin for `query_params` to diverge from the behavior of `params` when you modified this line but not line 101 below.

Reply:
Gotcha.  Yes that was intentional.  I figured it was safest to put this change behind the `:replace_query_params` option like we did in: https://github.com/beam-community/jsonapi/pull/282.

Reply:
Oh, good call. Can you add a code comment to the effect of “when we make this unconditional in the next major version we should apply the same treatment to query_params and params.”

Reply:
@mattpolzin Added the comment - let me know if that is sufficient (or feel free to change it!)